Documentation/upgrade.md

```diff
@@ -232,18 +269,42 @@ This is okay as long as the cluster health looks good and all monitors eventuall
 * **Use all nodes:** all storage nodes in the cluster will be managed by a single daemon set.
-Only the one daemon set will need to be edited to update the image version, then each OSD pod will need to be deleted so that a new pod will be created by the daemon set to take its place.
-* **Specify individual nodes:** each storage node specified in the cluster spec will be managed by its own individual replica set.
-Each of these replica sets will need to be edited to update the image version, then each OSD pod will need to be deleted so its replica set will start a new pod on the new version to replace it.
+Only the one daemon set will need to be edited to update the image version
+if upgrading from Rook 0.6.x or older, then each OSD pod will need to be deleted so that a new pod will be created by the daemon set to take its place.
+* **Specify individual nodes:** each storage node specified in the cluster spec will be managed by its own individual deployment.
+Each of these deployment will need to be edited to update the image version, then Kubernetes will start the update and a new pod still start with the new version to replace the old one.
+If upgrading from Rook 0.6.x or older, each OSD pod will be managed by a replica set instead. Each of these replica sets will need to be edited to update the image version, then each OSD pod will need to be deleted so its replica set will start a new pod on the new version to replace it.
+
+If upgrading from v0.6.x or older, you will also need to add a new field `ROOK_CLUSTER_ID` for both cases.
 
 In this example, we are going to walk through the case where `useAllNodes: true` was set in the cluster spec, so there will be a single daemon set managing all the OSD pods.
 
-Let's edit the image version of the pod template in the daemon set:
+#### Upgrading from 0.6.x or lower
+Let's edit the daemon set manifest to update the version and add the new field:
+```bash
+kubectl -n rook edit daemonset rook-ceph-osd
+```
+
+First update the image version to the new version of `rook/rook:master` as shown in the following snippet:
+```
+        image: rook/rook:master
+        imagePullPolicy: IfNotPresent
+        name: rook-ceph-osd
+```
+
+Second, add the `ROOK_CLUSTER_ID` variable under the `env` field. The value can be obtained by running the following (note that the UID of your cluster will be different):
 ```bash
-kubectl -n rook set image daemonset/rook-ceph-osd rook-ceph-osd=rook/rook:v0.5.1
+kubectl -n rook get cluster rook -o jsonpath='{.metadata.uid}{"\n"}'
+be1f9132-0147-11e8-9afe-080027c879a1
+```
+
+Add the UID value to the `ROOK_CLUSTER_ID` env: 
+```
+        - name: ROOK_CLUSTER_ID
+          value: be1f9132-0147-11e8-9afe-080027c879a1
 ```
 
-Once the daemon set is updated, we can begin deleting each OSD pod **one at a time** and verifying a new one comes up to replace it that is running the new version.
```

Comment:
why remove the emphasis?

Reply:
Because i already mentioned this before.. I can add it back.